### DISCUSSION TITLE
@midwayjs/http-proxy 支持手动使用

### DISCUSSION THREAD

**Discussion by SmartOrange** [midwayjs/midway#3376] [Ideas]
### The feature, motivation（功能、动机）

@midwayjs/http-proxy 目前只支持了 middleware 方式。 希望可以提供手动调用的能力。
类似
```
this.ctx.httpProxy.execProxy(url, { headers = {}, method = 'GET', data = {}});
```

**Comment by czy88840616** [comment]
手动怎么个用法。。。

**Reply by SmartOrange** [reply]
类似这样，context 注入一个 httpProxy 的方法。```this.ctx.httpProxy.execProxy(url, { headers = {}, method = 'GET', data = {}});```

**Reply by SmartOrange** [reply]
![image](https://github.com/midwayjs/midway/assets/15233745/98c9c5c0-6416-46af-aec4-7adfcf203846)

其实就是希望可以把红框里面的代码抽出来使用一下。 目前我是自己用 got 实现的。http-proxy 如果能提供的话，我就可以直接用http-proxy 了。 你看看有没有这个必要？

**Comment by czy88840616** [comment]
已经提供 enable 来关闭默认加载的中间件了。
